### PR TITLE
Paging details in listing responses

### PR DESCRIPTION
**Pull request by jackdos**
Resolves Issue #21

### CODE REVIEW

**Comment by jackdos** [comment]
I made a minor amendment to what the ticket (#21) said we should do, in that we specified that the parameters would be passed as an array of objects, but when implementing it seemed less verbose to make the parameters object essentially a string -> string map, so it's declared as a object with arbitrary keys and string values.

I've also used "patternProperties", which seems fine in JSON schema, but I've been struggling to get the OpenAPI spec to validate (quite a few tools seem to not like 3.1.0) or to generate the HTML, so I'm not 100% sure that that is as well supported in the API spec.

File: api/par-api.yaml [hunk 1]
```diff
@@ -122,14 +122,17 @@ paths:
...
         schema:
           type: integer
           default: 0
       - name: limit
         in: query
         description: Used for requesting paged responses, this defines the maximum
-          number of Business Rules to return.
+          number of Business Rules to return. If not specified, all results will be returned.
+          To explicitly request all results, set to a value < 0 (e.g. limit=-1). If set to 0
+          and empty list of results will be returned along with standard details of how many
```
Comment (carlwilson) [inline api/par-api.yaml:134]: Just wording but would "a count of matching results" be better here?

File: api/par-api.yaml [hunk 2]
```diff
@@ -1054,14 +1072,17 @@ paths:
...
           type: integer
           default: 0
       - name: limit
         in: query
         description: Used for requesting paged responses, this defines the maximum
-          number of FileFormats to return.
+          number of FileFormats to return. If not specified, all results will be returned.
+          To explicitly request all results, set to a value < 0 (e.g. limit=-1). If set to 0
+          and empty list of results will be returned along with standard details of how many
+          results actually matched.
```
Comment (carlwilson) [inline api/par-api.yaml:1085]: Similar here.

**Review by carlwilson — COMMENTED** [review]
This is fine TBH, see my niggly comments about wording though. These could be applied in a few places.

File: api/par-api.yaml (hunk 1 above)
Comment (jackdos) [reply api/par-api.yaml]: made the edit in all places (also fixed another typo that was copy-pasta'd throughout.

**Review by carlwilson — APPROVED** [review]

**Pull request merged**